### PR TITLE
Dbz 3805

### PR DESCRIPTION
**Pull request by yymoxiaochi**
1.  add origin message handle in PgOutputMessageDecoder
2. add origin name in ReplicationMessage and subclass
3. add origin name in SourceInfo
4. PostgresStreamingChangeEventSource will pass the origin name to sourceInfo

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java [hunk 1]
```diff
@@ -186,6 +189,10 @@ String tableName() {
         return tableName;
     }
 
+    String orignName() {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:192]:
```suggestion
    String originName() {
```

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java [hunk 2]
```diff
@@ -124,6 +126,7 @@ protected SourceInfo update(Lsn lsn, Instant commitTime, Long txId, TableId tabl
         if (tableId != null && tableId.table() != null) {
             this.tableName = tableId.table();
         }
+        this.orignName = originName;
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:129]:
```suggestion
        this.originName = originName;
```

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java [hunk 3]
```diff
@@ -81,6 +81,7 @@
     public static final String XMIN_KEY = "xmin";
     public static final String LSN_KEY = "lsn";
     public static final String LAST_SNAPSHOT_RECORD_KEY = "last_snapshot_record";
+    public static final String ORIGIN_NAME_KEY = "origin_name";
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:84]: I think just "origin" would suffice for the key.

**Review by gunnarmorling — COMMENTED** [review]
Thanks a lot, @yymoxiaochi! This is really interesting. A few more questions in additon to the comments inline:

* What's the "lifecycle" of the origin value? After receiving an origin message, will all change events subsequently emitted have the that origin, until another origin message comes in?
* Can we test this somehow? As per [this post](https://www.highgo.ca/2020/04/18/the-origin-in-postgresql-logical-decoding/), creating a publication should also trigger the creation (and subsequent usage) of an origin
* Can you update the docs, too? Specifically, the new field should be mentioned [here](https://debezium.io/documentation/reference/connectors/postgresql.html#postgresql-update-events)

**Comment by gunnarmorling** [comment]
Could you also squash the commits into one, rebase, and force-push? This will address the Octocat commit issue, and we don't have a merge commit which is preferred for the sake of a linear history. Thx!

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java (hunk 3 above)
Comment (yymoxiaochi) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/SourceInfo.java:84]: Hi,
1.  According to my tests, there is no origin value by default.  when we use
`select * from pg_replication_ORIGIN_session_setup ('origin_name')` later,
All subsequent operations on this connection will have the origin value attached. After `BEGIN`, each transaction.

2. I will do some test , but  the test of this project is too complicated. I need some time to study it.
3. I will update the docs.

Thanks.

**Comment by yymoxiaochi** [comment]
> Could you also squash the commits into one, rebase, and force-push? This will address the Octocat commit issue, and we don't have a merge commit which is preferred for the sake of a linear history. Thx!



> Could you also squash the commits into one, rebase, and force-push? This will address the Octocat commit issue, and we don't have a merge commit which is preferred for the sake of a linear history. Thx!

O.......M..........G..........    Do more research for git rebase T.T

**Comment by yymoxiaochi** [comment]
> Could you also squash the commits into one, rebase, and force-push? This will address the Octocat commit issue, and we don't have a merge commit which is preferred for the sake of a linear history. Thx!

Hi, could you close this PR as I will not rebase all submissions. I will do a new PR with the test.

**Comment by rk3rn3r** [comment]
@yymoxiaochi  you can rebase your branch and force-push the result. That will remove all the unwanted commits from your branch and from the PR. After cleaning up your branch's history, you can continue this PR if you like?

**Comment by gunnarmorling** [comment]
Hey @yymoxiaochi, how should we proceed with this one? Are you still planning to create a new PR? Would be nice to wrap this one up :) Thanks a lot!

**Comment by yymoxiaochi** [comment]
> Hey @yymoxiaochi, how should we proceed with this one? Are you still planning to create a new PR? Would be nice to wrap this one up :) Thanks a lot!

Hi, you can close this PR, I will create a new PR. thanks.

**Comment by gunnarmorling** [comment]
Gotcha; looking forward to the replacement. Thx!

**Comment by gunnarmorling** [comment]
Hey @yymoxiaochi, just was reminded of this. Do you still plan to send a new PR here?

**Pull request closed without merging**